res/res_respoke/respoke_endpoint.c

```diff
@@ -270,59 +350,64 @@ static int respoke_endpoint_apply(const struct ast_sorcery *sorcery, void *obj)
 	}
 
 	/* Determine if we need fresh state and to reconnect */
-	if (endpoint->state) {
-		if (can_reuse_state(endpoint->state, app, transport)) {
+	state = respoke_endpoint_state_retrieve(ast_sorcery_object_get_id(endpoint));
+	if (state) {
+		if (can_reuse_state(state, app, transport)) {
 			/* Nothing has changed that warrants us reconnecting */
 			ao2_ref(app, -1);
 			ao2_ref(transport, -1);
 			return 0;
 		}
-		ao2_ref(endpoint->state, -1);
+		/* Can't reuse it; get rid of the old state */
+		ao2_unlink(states, state);
+		ao2_ref(state, -1);
+		state = NULL;
```

Comment:
I don't think so.

Keep in mind that we would only drop the old state and create a new one if there was a relevant configuration change. If a failure then happens, your configuration is almost certainly suspect, at which point we have two options:

1. Use the old state with a configuration that no longer exists. That's bad for all sorts of reasons - what happens if Respoke sends us a call? How are we to answer it (or know what to do with it) if the endpoint/app/transport configuration is bogus?

2. Throw away the old state, and fail the configuration load.

Option 2 is a lot safer, and generally the approach we take in other places in Asterisk as well.